library/rsa.c

```diff
@@ -923,7 +923,7 @@ void mbedtls_rsa_set_padding( mbedtls_rsa_context *ctx, int padding, int hash_id
 
 size_t mbedtls_rsa_get_len( const mbedtls_rsa_context *ctx )
 {
-    return( mbedtls_mpi_size( &ctx->N ) );
+    return( ctx->len );
```

Comment:
Note for the future: I think this is fine for now, but in the future when we're allowed to change things, I think it would make more sense to not store the length and always compute it when needed. Besides saving a few bytes for RAM, it removes redundancy, and as N can be set in many places, it removes the risk of setting N without setting len.